src/app/modules/persistence-service/persistence.service.spec.ts

```diff
@@ -38,7 +40,184 @@ describe('PersistenceService', () => {
+      messagingService.publish('editor.dirtyStateChanged', { path: 'yet/another', dirty: true });
+      messagingService.publish('editor.dirtyStateChanged', { path: 'and/yet/another', dirty: true });
+      messagingService.publish('editor.dirtyStateChanged', { path: 'and/still/another', dirty: true });
+      messagingService.publish('editor.close', { id: 'some/file' }); // not open not dirty
+      messagingService.publish('editor.close', { id: 'yet/another' }); // not open, not dirty
+      messagingService.publish('editor.dirtyStateChanged', { path: 'and/yet/another', dirty: false }); // still open but not dirty
+      messagingService.publish('editor.save.completed', { id: 'and/still/another'}); // still open, but not dirty
+      persistenceService.copyResource('any', 'file'); // could be any other action that executes a pull
+      tick();
+
+       // then
+      const pullMatched = httpMock.match(pullMatcher)[0];
+      expect(pullMatched.request.body).toEqual(jasmine.objectContaining(
+        { resources: [ 'some/other/file', 'and/yet/another', 'and/still/another' ], dirtyResources: [ ] }));
+    })));
+
+  it('ensure that pull passes "resources" and "dirtyResources" without duplicates',
+      fakeAsync(inject([HttpTestingController, PersistenceService],
+                       (httpMock: HttpTestingController, persistenceService: PersistenceService) => {
+      // given
+      persistenceService.startSubscriptions();
+
+      // when
+      messagingService.publish('navigation.open', 'some/file');
+      messagingService.publish('navigation.open', 'some/other/file');
+      messagingService.publish('navigation.open', 'some/other/file');
+      messagingService.publish('editor.dirtyStateChanged', { path: 'and/yet/another', dirty: true });
+      messagingService.publish('editor.dirtyStateChanged', { path: 'and/yet/another', dirty: true });
+      persistenceService.copyResource('any', 'file'); // could be any other action that executes a pull
+      tick();
+
+       // then
+      const pullMatched = httpMock.match(pullMatcher)[0];
+      expect(pullMatched.request.body).toEqual(jasmine.objectContaining(
+        { resources: [ 'some/file', 'some/other/file' ], dirtyResources: [ 'and/yet/another' ] }));
+
+    })));
+
+  it('returns a conflict if backend reports a conflict in the backend', fakeAsync(inject([HttpTestingController, PersistenceService],
+    (httpMock: HttpTestingController, persistenceService: PersistenceService) => {
+      // given
+      const tclFilePath = 'path/to/file?.tcl';
+      const tclFileTarget = 'newpath/to/file?.tcl';
+      const message = `The file '${tclFileTarget}' already exists.`;
+      const expectedConflict = new Conflict(message);
+
+      // when
+      persistenceService.copyResource(tclFileTarget, tclFilePath)
+
+      // then
+      .then((result) => {
+        expect(result).toEqual(expectedConflict);
+      }, (response) => {
```

Comment:
```suggestion
      }, (error) => {
```
(or "rejected", "exception", etc., just for readability – calling it "response" may be confusing, especially in conjunction with the following test failure message…)